src/main/java/uk/gov/digital/ho/proving/income/audit/statistics/PassStatisticsResultsConsolidator.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.digital.ho.proving.income.audit.statistics;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.digital.ho.proving.income.audit.AuditResult;
+import uk.gov.digital.ho.proving.income.audit.AuditResultComparator;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class PassStatisticsResultsConsolidator {
+
+    private final AuditResultComparator resultComparator;
+    private final int cutoffDays;
+
+    PassStatisticsResultsConsolidator(AuditResultComparator resultComparator, @Value("${audit.history.cutoff.days}") int cutoffDays) {
+        this.resultComparator = resultComparator;
+        this.cutoffDays = cutoffDays;
+    }
+
+    List<AuditResult> consolidateResults(List<List<AuditResult>> resultsGroupedByNino) {
+        List<List<AuditResult>> separatedByCutoff = resultsGroupedByNino.stream()
+                                                                        .map(this::separateResultsByCutoff)
+                                                                        .flatMap(Collection::stream)
+                                                                        .collect(Collectors.toList());
+
+        return separatedByCutoff.stream()
+                                .map(this::earliestBestResult)
+                                .collect(Collectors.toList());
+    }
+
+    List<List<AuditResult>> separateResultsByCutoff(List<AuditResult> results) {
+        List<AuditResult> sortedByDate = sortByDate(results);
+        List<List<AuditResult>> groupedByCutoff = new ArrayList<>();
```

Comment:
Is it worth creating a type for List\<AuditResult\>?  It would make the code much more readable...

Reply:
You could add a date range to the new type, potentially making some of the grouping logic easier?